### ISSUE TITLE
Download Cache not work with latest version

### ISSUE THREAD

**Issue by tobka777** [devonfw/ide#1293]
### Expected behavior

As a tool user, I want to have every time the latest version so that I can work with the newest features.

### Actual behavior

We have a "Download Cache" for devonfw-ide. For this we download tools and put them under `~/Downloads/devonfw-ide/«tool»-«version»[-«edition»].«extension»`.

We sometimes have as «version» simply latest and this means semantically that it is the latest version. But this does not make the filename and the file content permanently stable. So in this case it is wrong that if the file already exists, we don't re-download it.

In other words: We are not allowed to use latest as version, but we have to solve the updater problems. Our scripts must then resolve something like latest to the correct version and then download that specifically to a file with the version on it. This concerns at least pip and docker-desktop.

### Related/Dependent Issues

* https://github.com/devonfw/ide/issues/1170

### Affected version:

* OS: Windows/Linux/Mac

**Comment by hohwille** [comment]
So `doDownload` should not save or reuse `latest` version in download cache folder.
For docker, I will create a separate story to improve the UrlUpdater.
